### PR TITLE
[DI] Fix XmlFileLoader bad error message

### LINKED ISSUE #35886: [DI] XmlFileLoader bad error message
Today we debugged an error message coming from `XmlFileLoader` in the DI component:

```
File some.xml does not contain valid XML, it is empty.
```

Coming from

https://github.com/symfony/symfony/blob/421c7f8fcd42aa64d1d7d6adee0558993ede9a41/src/Symfony/Component/Config/Util/XmlUtils.php#L125-L128

The file is not empty :) removing the `@` reveals the true error: 

```
PHP Warning:  file_get_contents(file.ext): failed to open stream: Permission denied 
```

So if `$content` is false (before trim casts to string) we cant assume an empty file. Ideally the original errors bubbles up.

### PR DESCRIPTION
**Pull request by przemyslaw-bogusz**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #35886 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by AdrianDeAnda — APPROVED** [review]

File: src/Symfony/Component/Config/Util/XmlUtils.php [hunk 1]
```diff
@@ -122,7 +122,16 @@ public static function parse($content, $schemaOrCallable = null)
      */
     public static function loadFile($file, $schemaOrCallable = null)
     {
-        $content = @file_get_contents($file);
+        if (!is_file($file)) {
+            throw new \InvalidArgumentException(sprintf('Resource %s is not a file.', $file));
```
Comment (fabpot) [inline src/Symfony/Component/Config/Util/XmlUtils.php]: Can you quote the file name `"%"`? Same below (and if you can also fix the one line 136, that would be awesome).

File: src/Symfony/Component/Config/Util/XmlUtils.php [hunk 2]
```diff
@@ -122,7 +122,16 @@ public static function parse($content, $schemaOrCallable = null)
...
+        }
+
+        if (!is_readable($file)) {
+            throw new \InvalidArgumentException(sprintf('File %s is not readable.', $file));
+        }
+
+        $content = file_get_contents($file);
```
Comment (fabpot) [inline src/Symfony/Component/Config/Util/XmlUtils.php]: I would keep `@` here.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Config/Util/XmlUtils.php (hunk 2 above)
Comment (przemyslaw-bogusz) [reply src/Symfony/Component/Config/Util/XmlUtils.php]: done

File: src/Symfony/Component/Config/Util/XmlUtils.php (hunk 1 above)
Comment (przemyslaw-bogusz) [reply src/Symfony/Component/Config/Util/XmlUtils.php]: done

File: src/Symfony/Component/Config/Tests/Util/XmlUtilsTest.php [hunk 3]
```diff
@@ -20,6 +20,31 @@ public function testLoadFile()
...
+        } catch (\InvalidArgumentException $e) {
+            $this->assertStringContainsString('is not a file', $e->getMessage());
+        }
+
+        try {
+            if ('\\' === \DIRECTORY_SEPARATOR) {
+                $this->markTestSkipped('chmod is not supported on Windows');
+            }
+            chmod($fixtures.'not_readable.xml', 000);
+            XmlUtils::loadFile($fixtures.'not_readable.xml');
```
Comment (przemyslaw-bogusz) [inline src/Symfony/Component/Config/Tests/Util/XmlUtilsTest.php]: @fabpot I've also changed this filename from `test.xml` to `not_readable.xml`.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @przemyslaw-bogusz.

**Pull request merged**